rca/static_src/javascript/components/anchor-nav.js

```diff
@@ -0,0 +1,78 @@
+import 'intersection-observer';
+import scrollama from 'scrollama';
+
+class AnchorNav {
+    static selector() {
+        return '[data-anchor-nav]';
+    }
+
+    constructor(node) {
+        this.node = node;
+        this.allLinks = this.node.querySelectorAll('a[href^="#"]');
+        this.scrollamaInit();
+    }
+
+    scrollamaInit() {
+        // instantiate the scrollama
+        const scroller = scrollama();
+
+        // setup the instance, pass callback functions
+        scroller
+            .setup({
+                step: '.anchor-heading',
+                offset: 0.05,
+            })
+            .onStepEnter((el) => {
+                if (el.direction === 'down') {
+                    this.handleDown(el);
+                }
+            })
+            .onStepExit((el) => {
+                if (el.direction === 'up') {
+                    this.handleUp(el);
+                }
+            });
+
+        // setup resize event
+        window.addEventListener('resize', scroller.resize);
```

Comment:
We could use [`ResizeObserver`](https://web.dev/resize-observer/) here, but the performance gains may be minimal…